app_dart/lib/src/service/luci_build_service.dart

```diff
@@ -86,7 +86,7 @@ class LuciBuildService {
   /// Returns an Iterable of try Buildbucket [Build]s for a given [PullRequest].
   Future<Iterable<Build>> getTryBuildsByPullRequest(
     github.PullRequest pullRequest,
-  ) {
```

Comment:
This isn't necessary.

https://dart-lang.github.io/linter/lints/unnecessary_await_in_return.html

Reply:
But I am not awaiting a return.

Reply:
@CaseyHillers you are right I just proved it through a deployment. Something is up, because I see this working when I have tested it.

Reply:
See the example:

```
GOOD:
Future<int> f2() {
  return future;
}
```

These lints aren't intended to explain the concept, but to enforce best practices. We should add the lint so this PR fails on additions like this.

See https://dart.dev/guides/libraries/library-tour#future for more info on why, but whenever possible, the callers of methods should do await, not the method itself.

Reply:
Man I think we do this a lot throughout the code too. But this is good. I will add this and do some cleanup. Thank you for the extra info here.

Reply:
Okay I am going to disagree with you on this change. Adding the linter rule does not flag the async keyword. And it appears to be working every time now except for when a PR is first opened. This is the fix.

Reply:
FYI I created https://github.com/flutter/cocoon/pull/2644. Seems like several analysis options are failing at HEAD due to this check not running in CI.

\cc @christopherfujino have you seen this issue before? I don't see anything functionally wrong that would require async to be in the method, as this is only returning a future.